File: src/user_examples/cs_user_boundary_conditions-gas_3ptchem.f90
!-------------------------------------------------------------------------------

!VERS

! This file is part of Code_Saturne, a general-purpose CFD tool.
!
! Copyright (C) 1998-2020 EDF S.A.
!
! This program is free software; you can redistribute it and/or modify it under
! the terms of the GNU General Public License as published by the Free Software
! Foundation; either version 2 of the License, or (at your option) any later
! version.
!
! This program is distributed in the hope that it will be useful, but WITHOUT
! ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
! FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
! details.
!
! You should have received a copy of the GNU General Public License along with
! this program; if not, write to the Free Software Foundation, Inc., 51 Franklin
! Street, Fifth Floor, Boston, MA 02110-1301, USA.

!-------------------------------------------------------------------------------

!===============================================================================
! Function:
! ---------
!> \file  cs_user_boundary_conditions-gas_3pthem.f90
!> \brief Example of cs_user_boundary_conditions subroutine.f90 for 3 PTHEM gas
!
!-------------------------------------------------------------------------------

!-------------------------------------------------------------------------------
! Arguments
!______________________________________________________________________________.
!  mode           name          role                                           !
!______________________________________________________________________________!
!> \param[in]     nvar          total number of variables
!> \param[in]     nscal         total number of scalars
!> \param[out]    icodcl        boundary condition code:
!>                               - 1 Dirichlet
!>                               - 2 Radiative outlet
!>                               - 3 Neumann
!>                               - 4 sliding and
!>                                 \f$ \vect{u} \cdot \vect{n} = 0 \f$
!>                               - 5 smooth wall and
!>                                 \f$ \vect{u} \cdot \vect{n} = 0 \f$
!>                               - 6 rough wall and
!>                                 \f$ \vect{u} \cdot \vect{n} = 0 \f$
!>                               - 9 free inlet/outlet
!>                                 (input mass flux blocked to 0)
!> \param[in]     itrifb        indirection for boundary faces ordering
!> \param[in,out] itypfb        boundary face types
!> \param[out]    izfppp        boundary face zone number
!> \param[in]     dt            time step (per cell)
!> \param[in,out] rcodcl        boundary condition values:
!>                               - rcodcl(1) value of the dirichlet
!>                               - rcodcl(2) value of the exterior exchange
!>                                 coefficient (infinite if no exchange)
!>                               - rcodcl(3) value flux density
!>                                 (negative if gain) in w/m2 or roughness
!>                                 in m if icodcl=6
!>                                 -# for the velocity \f$ (\mu+\mu_T)
!>                                    \gradt \, \vect{u} \cdot \vect{n}  \f$
!>                                 -# for the pressure \f$ \Delta t
!>                                    \grad P \cdot \vect{n}  \f$
!>                                 -# for a scalar \f$ cp \left( K +
!>                                     \dfrac{K_T}{\sigma_T} \right)
!>                                     \grad T \cdot \vect{n} \f$
!_______________________________________________________________________________

subroutine cs_f_user_boundary_conditions &
 ( nvar   , nscal  ,                                              &
   icodcl , itrifb , itypfb , izfppp ,                            &
   dt     ,                                                       &
   rcodcl )

!===============================================================================

!===============================================================================
! Module files
!===============================================================================

use paramx
use numvar
use optcal
use cstphy
use cstnum
use entsor
use parall
use period
use ppppar
use ppthch
use coincl
use cpincl
use ppincl
use ppcpfu
use atincl
use ctincl
use cs_fuel_incl
use mesh
use field

!===============================================================================

implicit none

! Arguments

integer          nvar   , nscal

integer          icodcl(nfabor,nvar)
integer          itrifb(nfabor), itypfb(nfabor)
integer          izfppp(nfabor)

double precision dt(ncelet)
double precision rcodcl(nfabor,nvar,3)

! Local variables

!< [loc_var_dec]
integer          iel, ifac, izone, ii
integer          ilelt, nlelt

double precision uref2, d2s3
double precision xkent, xeent
double precision dp, rho, S0, Kadm, radm, madm, Qadm, Padm, Ploc
double precision, dimension(:), pointer :: crom

integer, allocatable, dimension(:) :: lstelt
!< [loc_var_dec]

!===============================================================================

!===============================================================================
! Initialization
!===============================================================================

allocate(lstelt(nfabor))  ! temporary array for boundary faces selection

d2s3 = 2.d0/3.d0

!===============================================================================
! Assign boundary conditions to boundary faces here

! For each subset:
! - use selection criteria to filter boundary faces of a given subset
! - loop on faces from a subset
!   - set the boundary condition for each face
!===============================================================================

!   Definition of a fuel flow inlet for each face of color 11

!< [example_1]
call getfbr('11', nlelt, lstelt)
!==========

do ilelt = 1, nlelt

  ifac = lstelt(ilelt)

!   Type of pre-defined boundary condidition (see above)
  itypfb(ifac) = ientre

!   Zone number (arbitrary number between 1 and n)
  izone = 1

!   Allocation of the actual face to the zone
  izfppp(ifac) = izone

!   Indicating the inlet as a fuel flow inlet
  ientfu(izone) = 1

!   Inlet Temperature in K
  tinfue = 436.d0

!   The incoming fuel flow refers to:
!   a) a massflow rate   -> iqimp()  = 1
  iqimp(izone)  = 1
  qimp(izone)   = 2.62609d-4 / 72.d0
!
!   b) an inlet velocity -> iqimp()  = 0
  rcodcl(ifac,iu,1) = 0.d0
  rcodcl(ifac,iv,1) = 0.d0
  rcodcl(ifac,iw,1) = 21.47d0
! ATTENTION: If iqimp()  = 1 the direction vector of the massfow has
!            to begiven here.
!
!   Boundary conditions of turbulence
  icalke(izone) = 1
!
!    - If ICALKE = 0 the boundary conditions of turbulence at
!      the inlet are calculated as follows:

  if(icalke(izone).eq.0) then

    uref2 = rcodcl(ifac,iu,1)**2                           &
           +rcodcl(ifac,iv,1)**2                           &
           +rcodcl(ifac,iw,1)**2
    uref2 = max(uref2,1.d-12)
    xkent  = epzero
    xeent  = epzero

    if    (itytur.eq.2) then

      rcodcl(ifac,ik,1)  = xkent
      rcodcl(ifac,iep,1) = xeent

    elseif(itytur.eq.3) then

      rcodcl(ifac,ir11,1) = d2s3*xkent
      rcodcl(ifac,ir22,1) = d2s3*xkent
      rcodcl(ifac,ir33,1) = d2s3*xkent
      rcodcl(ifac,ir12,1) = 0.d0
      rcodcl(ifac,ir13,1) = 0.d0
      rcodcl(ifac,ir23,1) = 0.d0
      rcodcl(ifac,iep,1)  = xeent

    elseif (iturb.eq.50) then

      rcodcl(ifac,ik,1)   = xkent
      rcodcl(ifac,iep,1)  = xeent
      rcodcl(ifac,iphi,1) = d2s3
      rcodcl(ifac,ifb,1)  = 0.d0

    elseif (iturb.eq.60) then

      rcodcl(ifac,ik,1)   = xkent
      rcodcl(ifac,iomg,1) = xeent/cmu/xkent

    elseif (iturb.eq.70) then

      rcodcl(ifac,inusa,1) = cmu*xkent**2/xeent

    endif

  endif
!
!    - If ICALKE = 1 the boundary conditions of turbulence at
!      the inlet refer to both, a hydraulic diameter and a
!      reference velocity.
!
  dh(izone)     = 0.032d0
!
!    - If ICALKE = 2 the boundary conditions of turbulence at
!      the inlet refer to a turbulence intensity.
!
  xintur(izone) = 0.d0

enddo
!< [example_1]

!   Definition of an air flow inlet for each face of color 21

!< [example_2]
call getfbr('21', nlelt, lstelt)
!==========

do ilelt = 1, nlelt

  ifac = lstelt(ilelt)

!   Type of pre-defined boundary condidition (see above)
  itypfb(ifac) = ientre

!   Zone number (arbitrary number between 1 and n)
  izone = 2

!   Allocation of the actual face to the zone
  izfppp(ifac) = izone

!   Indicating the inlet as a air flow inlet
  ientox(izone) = 1

!   Inlet Temperature in K
  tinoxy = 353.d0

!   The inflowing fuel flow refers to:
!   a) a massflow rate   -> iqimp()  = 1
  iqimp(izone)  = 1
  qimp(izone)   = 4.282d-3 / 72.d0
!
!   b) an inlet velocity -> iqimp()  = 0
  rcodcl(ifac,iu,1) = 0.d0
  rcodcl(ifac,iv,1) = 0.d0
  rcodcl(ifac,iw,1) = 0.097d0
! ATTENTION: If iqimp()  = 1 the direction vector of the massfow has
!            to be given here.

!   Boundary conditions of turbulence
  icalke(izone) = 1
!
!    - If ICALKE = 0 the boundary conditions of turbulence at
!      the inlet are calculated as follows:

  if(icalke(izone).eq.0) then

    uref2 = rcodcl(ifac,iu,1)**2                           &
           +rcodcl(ifac,iv,1)**2                           &
           +rcodcl(ifac,iw,1)**2
    uref2 = max(uref2,1.d-12)
    xkent  = epzero
    xeent  = epzero

    if    (itytur.eq.2) then

      rcodcl(ifac,ik,1)  = xkent
      rcodcl(ifac,iep,1) = xeent

    elseif(itytur.eq.3) then

      rcodcl(ifac,ir11,1) = d2s3*xkent
      rcodcl(ifac,ir22,1) = d2s3*xkent
      rcodcl(ifac,ir33,1) = d2s3*xkent
      rcodcl(ifac,ir12,1) = 0.d0
      rcodcl(ifac,ir13,1) = 0.d0
      rcodcl(ifac,ir23,1) = 0.d0
      rcodcl(ifac,iep,1)  = xeent

    elseif (iturb.eq.50) then

      rcodcl(ifac,ik,1)   = xkent
      rcodcl(ifac,iep,1)  = xeent
      rcodcl(ifac,iphi,1) = d2s3
      rcodcl(ifac,ifb,1)  = 0.d0

    elseif (iturb.eq.60) then

      rcodcl(ifac,ik,1)   = xkent
      rcodcl(ifac,iomg,1) = xeent/cmu/xkent

    elseif (iturb.eq.70) then

      rcodcl(ifac,inusa,1) = cmu*xkent**2/xeent

    endif

  endif
!
!    - If ICALKE = 1 the boundary conditions of turbulence at
!      the inlet refer to both, a hydraulic diameter and a
!      reference velocity.
!
  dh(izone)     = 0.218d0
!
!    - If ICALKE = 2 the boundary conditions of turbulence at
!      the inlet refer to a turbulence intensity.
!
  xintur(izone) = 0.d0
!
enddo
!< [example_2]

!  Definition of a wall for each face of color 51 up to 59

!< [example_3]
call getfbr('51 to 59', nlelt, lstelt)
!==========

do ilelt = 1, nlelt

  ifac = lstelt(ilelt)

!   Type of pre-defined boundary condidition (see above)
  itypfb(ifac)   = iparoi

!   Zone number (arbitrary number between 1 and n)
  izone = 3

!   Allocation of the actual face to the zone
  izfppp(ifac) = izone

enddo
!< [example_3]


!  Definition of an exit for each face of color 91

!< [example_4]
call getfbr('91', nlelt, lstelt)
!==========

do ilelt = 1, nlelt

  ifac = lstelt(ilelt)

!   Type of pre-defined boundary condidition (see above)
  itypfb(ifac)   = isolib

!   Zone number (arbitrary number between 1 and n)
  izone = 4

!   Allocation of the actual face to the zone
  izfppp(ifac) = izone

enddo
!< [example_4]

!  Definition of symmetric boundary conditions for each
!  face of color 41 and 4.

!< [example_5]
call getfbr('41 or 4', nlelt, lstelt)
!==========

do ilelt = 1, nlelt

  ifac = lstelt(ilelt)

!   Type of pre-defined boundary condidition (see above)
  itypfb(ifac)   = isymet

!   Zone number (arbitrary number between 1 and n)
  izone = 5

!   Allocation of the actual face to the zone
  izfppp(ifac) = izone

enddo
!< [example_5]

! Definition of an air supply for each face of color 61

!< [example_6]
call getfbr('61', nlelt, lstelt)
!==========

call field_get_val_s(icrom, crom)

! Head losses in the ventilation pipe (often from exp data)

! nominal room pressure
Ploc= 51.d0 ! Pa
! vent surface
S0 = acos(-1.d0) * (75.d-3)**2 ! m2
! density
radm = (P0+ploc)/(cs_physical_constants_r*tinoxy/wmolg(2))
! nominal vent pressure
Padm = 119.d0 ! Pa
! nominal flow rate
Qadm = 504.d0 ! m3/h
madm = Qadm*radm/3600.d0
! head loss
Kadm = 2.d0*radm*(S0/madm)**2 * abs(Ploc-Padm)

! pressure gradient (Pther computed if ipthrm = 1)
dp   = (Pther - P0) - Padm

do ilelt = 1, nlelt

  ifac = lstelt(ilelt)
  iel  = ifabor(ifac)

  ! Density
  if (dp.gt.0.d0) then
    rho = crom(iel)
  else
    rho = radm
  endif

  ! Mass flow rate (opposed to the boundary face normal vector)
  madm = - sign(1.d0,dp) * s0 * sqrt(2.d0*rho/kadm*abs(dp))

  ! Type of pre-defined boundary condidition (see above)
  itypfb(ifac) = i_convective_inlet

  ! Zone number (arbitrary number between 1 and n)
  izone = 6

  ! Allocation of the actual face to the zone
  izfppp(ifac) = izone

  ! Indicating the inlet as a air flow inlet
  ientox(izone) = 1

  ! Inlet Temperature in K
  tinoxy = t0

  ! The inflowing fuel flow refers to:
  ! a) a massflow rate   -> iqimp()  = 1
  iqimp(izone)  = 1
  qimp(izone)   = madm

  ! b) an inlet velocity -> iqimp()  = 0
  rcodcl(ifac,iu,1) = 1.d0
  rcodcl(ifac,iv,1) = 0.d0
  rcodcl(ifac,iw,1) = 0.d0

  ! Boundary conditions of turbulence
  icalke(izone) = 1

  ! - If ICALKE = 0 the boundary conditions of turbulence at
  !   the inlet are calculated as follows:
  if(icalke(izone).eq.0) then

    uref2 = rcodcl(ifac,iu,1)**2                           &
           +rcodcl(ifac,iv,1)**2                           &
           +rcodcl(ifac,iw,1)**2
    uref2 = max(uref2,1.d-12)
    xkent  = epzero
    xeent  = epzero

    if    (itytur.eq.2) then

      rcodcl(ifac,ik,1)  = xkent
      rcodcl(ifac,iep,1) = xeent

    elseif(itytur.eq.3) then

      rcodcl(ifac,ir11,1) = d2s3*xkent
      rcodcl(ifac,ir22,1) = d2s3*xkent
      rcodcl(ifac,ir33,1) = d2s3*xkent
      rcodcl(ifac,ir12,1) = 0.d0
      rcodcl(ifac,ir13,1) = 0.d0
      rcodcl(ifac,ir23,1) = 0.d0
      rcodcl(ifac,iep,1)  = xeent

    elseif (iturb.eq.50) then

      rcodcl(ifac,ik,1)   = xkent
      rcodcl(ifac,iep,1)  = xeent
      rcodcl(ifac,iphi,1) = d2s3
      rcodcl(ifac,ifb,1)  = 0.d0

    elseif (iturb.eq.60) then

      rcodcl(ifac,ik,1)   = xkent
      rcodcl(ifac,iomg,1) = xeent/cmu/xkent

    elseif (iturb.eq.70) then

      rcodcl(ifac,inusa,1) = cmu*xkent**2/xeent

    endif

  endif

  ! - If ICALKE = 1 the boundary conditions of turbulence at
  !   the inlet refer to both, a hydraulic diameter and a
  !   reference velocity.
  dh(izone)     = 0.218d0

  ! - If ICALKE = 2 the boundary conditions of turbulence at
  !   the inlet refer to a turbulence intensity.
  xintur(izone) = 0.d0

enddo
!< [example_6]

!--------
! Formats
!--------

!----
! End
!----

deallocate(lstelt)  ! temporary array for boundary faces selection

return
end subroutine cs_f_user_boundary_conditions

File: src/user_examples/cs_user_boundary_conditions-gas_3ptchem.c
/*============================================================================
 * User definition of boundary conditions.
 *============================================================================*/

/* VERS */

/*
  This file is part of code_saturne, a general-purpose CFD tool.

  Copyright (C) 1998-2024 EDF S.A.

  This program is free software; you can redistribute it and/or modify it under
  the terms of the GNU General Public License as published by the Free Software
  Foundation; either version 2 of the License, or (at your option) any later
  version.

  This program is distributed in the hope that it will be useful, but WITHOUT
  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
  FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
  details.

  You should have received a copy of the GNU General Public License along with
  this program; if not, write to the Free Software Foundation, Inc., 51 Franklin
  Street, Fifth Floor, Boston, MA 02110-1301, USA.
*/

/*----------------------------------------------------------------------------*/

#include "cs_defs.h"

/*----------------------------------------------------------------------------
 * Standard C library headers
 *----------------------------------------------------------------------------*/

#include <assert.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

/*----------------------------------------------------------------------------
 * Local headers
 *----------------------------------------------------------------------------*/

#include "cs_headers.h"

/*----------------------------------------------------------------------------*/

BEGIN_C_DECLS

/*----------------------------------------------------------------------------*/
/*!
 * \file cs_user_boundary_conditions.c
 *
 * \brief User functions for boundary condition definitions.
 */
/*----------------------------------------------------------------------------*/

/*=============================================================================
 * User function definitions
 *============================================================================*/

/*----------------------------------------------------------------------------*/
/*!
 * \brief Set boundary conditions to be applied.
 *
 * This function is called just before \ref cs_user_finalize_setup, and
 * boundary conditions can be defined in either of those functions,
 * depending on whichever is considered more readable or practical for a
 * given use.
 *
 * \param[in, out]  domain  pointer to a cs_domain_t structure
 */
/*----------------------------------------------------------------------------*/

void
cs_user_boundary_conditions_setup(cs_domain_t  *domain)
{
  CS_UNUSED(domain);

  /*! [setup_inlet] */
  const cs_zone_t *z = cs_boundary_zone_by_name("inlet");

  cs_real_t q_burner = cs_notebook_parameter_value_by_name("q_burner");

  cs_real_t q_inlet = q_burner / 360.;                             /* W */
  cs_real_t m_inlet = q_inlet / cs_glob_combustion_model->pcigas;  /* kg/s */
  cs_real_t r_inlet = 0.15;                                        /* m */

  cs_boundary_conditions_open_set_mass_flow_rate_by_value(z, m_inlet);

  cs_boundary_conditions_inlet_set_turbulence_hyd_diam(z, 2.*r_inlet);
  /*! [setup_inlet] */
}

/*----------------------------------------------------------------------------*/
/*!
 * \brief User definition of boundary conditions
 *
 * \param[in, out]  domain   pointer to a cs_domain_t structure
 * \param[in, out]  bc_type  boundary face types
 *
 * The icodcl and rcodcl arrays are pre-initialized based on default
 * and GUI-defined definitions, and may be modified here.
 *
 * For a given variable field f, and a given face "face_id", these arrays
 * may be used as follows:
 *
 * - Boundary condition type code given at:
 *   f->bc_coeffs->icodcl[face_id]
 *
 * - Dirichlet value defined at:
 *   f->bc_coeffs->rcodcl1[face_id]
 *
 * - Interior exchange coefficient (infinite if no exchange) at:
 *   f->bc_coeffs->rcodcl2[face_id]
 *
 * - Flux density defined at:
 *   f->bc_coeffs->rcodcl3[face_id]
 *
 * For vector or tensor fields, these arrays are not interleaved,
 * so for a given face "face_id" and field component "comp_id", access
 * is as follows (where n_b_faces is domain->mesh->n_b_faces):
 *
 *   f->bc_coeffs->rcodcl1[n_b_faces*comp_id + face_id]\n
 *   f->bc_coeffs->rcodcl2[n_b_faces*comp_id + face_id]\n
 *   f->bc_coeffs->rcodcl3[n_b_faces*comp_id + face_id]\n\n
 *
 * Only the icodcl code values from the first component are used in the case
 * of vector or tensor fields, so the icodcl values can be defined as for
 * a scalar.
 */
/*----------------------------------------------------------------------------*/

void
cs_user_boundary_conditions(cs_domain_t  *domain,
                            int           bc_type[])
{
  /*! [loc_var_dec] */

  const cs_lnum_t *b_face_cells = domain->mesh->b_face_cells;
  const cs_real_3_t *cdgfbo
    = (const cs_real_3_t *)(domain->mesh_quantities->b_face_cog);
  const cs_real_t *gxyz = cs_glob_physical_constants->gravity;

  const cs_fluid_properties_t *fp = cs_glob_fluid_properties;
  cs_combustion_model_t  *cm = cs_glob_combustion_model;

  const int kbmasf = cs_field_key_id("boundary_mass_flux_id");
  const int iflmab =  cs_field_get_key_int(CS_F_(vel), kbmasf);
  const cs_real_t *bmasfl = cs_field_by_id(iflmab)->val;

  /* Oxydant temperature needed by the combustion model
     when there is no oxydant inlet */

  cs_real_t coefg[CS_COMBUSTION_GAS_MAX_GLOBAL_SPECIES];
  for (int icg = 0; icg < CS_COMBUSTION_GAS_MAX_GLOBAL_SPECIES; icg++)
    coefg[icg] = 0;
  coefg[1] = 1;

  cm->tinoxy = fp->t0;
  cm->hinoxy = cs_gas_combustion_t_to_h(coefg, cm->tinoxy);

  /*! [loc_var_dec] */

  /* Fuel inlet */

  /*! [inlet] */
  {
    const cs_zone_t *z = cs_boundary_zone_by_name("inlet");

    /* Fuel flow inlet type */
    cs_glob_bc_pm_info->ientfu[z->id] = 1;

    /* Inlet Temperature in K */
    cm->gas->tinfue = fp->t0;
  }
  /*! [inlet] */

  /* Open boundary: free inlet/outlet with backflow conditions.

     outlet: zero flux for velocity and temperature, prescribed pressure. */

  /*! [open] */
  {
    const cs_real_3_t *cvar_vel = (const cs_real_3_t *)CS_F_(vel)->val;

    cs_real_t *brom = CS_F_(rho_b)->val;

    int *p_icodcl = CS_F_(p)->bc_coeffs->icodcl;
    cs_real_t *p_rcodcl1 = CS_F_(p)->bc_coeffs->rcodcl1;
    cs_real_t *fm_rcodcl1 = CS_F_(fm)->bc_coeffs->rcodcl1;
    cs_real_t *fp2m_rcodcl1 = CS_F_(fp2m)->bc_coeffs->rcodcl1;
    cs_real_t *h_rcodcl1 = NULL;
    if (cs_glob_physical_model_flag[CS_COMBUSTION_3PT] == 1) {
      h_rcodcl1 = CS_F_(h)->bc_coeffs->rcodcl1;
    }

    const cs_real_t xdh = 3.0;     /* acos(-1.d0) * 1 ( domain circumference) */
    const cs_real_t xitur = 0.02;  /* turbulence intensity */

    /* Loop on zone faces */

    const cs_zone_t *z = cs_boundary_zone_by_name("open");

    for (cs_lnum_t idx = 0; idx < z->n_elts; idx++) {
      const cs_lnum_t face_id = z->elt_ids[idx];

      bc_type[face_id] = CS_FREE_INLET;

      /* Imposed pressure */

      const cs_real_t pimp
        = fp->p0 + fp->ro0*cs_math_3_distance_dot_product(fp->xyzp0,
                                                          cdgfbo[face_id],
                                                          gxyz);
      p_icodcl[face_id] = 1;
      p_rcodcl1[face_id] = pimp;

      /* Outlet: Neumann conditions by defauls */

      if (bmasfl[face_id] > 0)
        continue;

      /* Inlet (backflow): Dirichlet conditions */

      const cs_lnum_t c_id = b_face_cells[face_id];

      cs_real_t uref2 = cs_math_3_square_norm(cvar_vel[c_id]);
      uref2 = cs_math_fmax(uref2, cs_math_epzero);

      cs_turbulence_bc_inlet_turb_intensity(face_id, uref2, xitur, xdh);

      /* Mixture fraction */
      fm_rcodcl1[face_id] = 0;

      /* Variance */
      fp2m_rcodcl1[face_id] = 0;

      /* Enthalpy */
      if (h_rcodcl1 != NULL)
        h_rcodcl1[face_id] = cm->hinoxy;

      /* Soot */
      if (cm->isoot == 1) {
        CS_F_(fsm)->bc_coeffs->rcodcl1[face_id] = 0;
        CS_F_(npm)->bc_coeffs->rcodcl1[face_id] = 0;
      }

      /* Density */
      brom[face_id] = fp->p0/(  cs_physical_constants_r
                              * cm->tinoxy/cm->gas->wmolg[1]);
    }
  }
  /*! [open] */
}

/*----------------------------------------------------------------------------*/

END_C_DECLS
